LICENSE-DETAILS.md

```diff
@@ -0,0 +1,80 @@
+
+This commit is the result of running a spellchecker on the code base.
+While it is larger than 10 lines of code, it is very similar to the example commit mentioned above which mechanically removed the full stop from the end of various sentences in the codebase.
+As such, it does not seem "sufficiently original" to be copyrightable.
+
+### [`f907b891da1641034f0603b0c6bc00b7aa4d1f4a`](https://github.com/llvm/llvm-project/commit/f907b891da1641034f0603b0c6bc00b7aa4d1f4a)
+
+This commit fixes hyphenation of words mostly in code comments or documentation.
+Again, while it is larger than 10 lines of code, it is very similar to the example commit mentioned above and does not seem "sufficiently original" to be copyrightable.
+
+### [`bf19e0a5561593d3d25924693a20a9bbe7771a3f`](https://github.com/llvm/llvm-project/commit/bf19e0a5561593d3d25924693a20a9bbe7771a3f)
+
+This commit fixes a comparison to work properly with the MSVC compiler.
+As the total diff is only 4 lines, it does not meet the established threshold of originality.
+
+### [`49c758b769b9a787d415a9c2ce0e40fa0e482412`](https://github.com/llvm/llvm-project/commit/49c758b769b9a787d415a9c2ce0e40fa0e482412)
+
+This commit changes some uses of raw pointers to use `unique_ptr`.
+While the commit is larger than 10 lines of code, it is reverted in the next commit which landed the following day.
+As such, the combination of this commit and the following commit are zero lines of code changed in the codebase.
+
+### [`4e69e29a72a1ffcbf755f13ed909b51cfbcafd60`](https://github.com/llvm/llvm-project/commit/4e69e29a72a1ffcbf755f13ed909b51cfbcafd60)
+
+This commit is a mechanical revert of the prior commit.
+
+### [`8710d9831b1a78e3ecb1f49da24447ee27f73096`](https://github.com/llvm/llvm-project/commit/8710d9831b1a78e3ecb1f49da24447ee27f73096)
+
+This commit changes a `struct` definition to be a `class`.
+As the total diff is only 2 lines, it does not meet the established threshold of originality.
+
+### [`1c419ff50d35c4cab442f5a1c8f5e82812921633`](https://github.com/llvm/llvm-project/commit/1c419ff50d35c4cab442f5a1c8f5e82812921633)
+
+This commit fixes an assertion in the code which had not been correctly updated in a prior change.
+As the total diff is only 6 lines of code (excluding changes to APInt unit tests which we did not port to Rust), it does not meet the established threshold of originality.
+
+# Conclusion
+
+The original LLVM code appears to be available as Apache 2 with LLVM Exception and so our port of this code is licensed as such.
+
+A few additional patches (code cleanups and backports of upstream LLVM changes) have been made on top of the initial port.
+The authors of these patches have also agreed to allow their code to be used under the Apache 2 with LLVM Exception license (backports from LLVM only include changes made after LLVM's licensing switch on 2019-01-19).
```

Comment:
This is no longer relevant, right? Since the backporting was replaced with moving the whole port forward?

Reply:
Yes, the LLVM backports themselves weren't used as you just moved the port forward to include them. However, there were two other commits (https://github.com/rust-lang/rust/commit/964ba2a6e731f79d99248d1f0e67eac17e69e368 and https://github.com/rust-lang/rust/commit/a66d7b2001def5ab87e99d661e31890660f38374) that we also got sign offs for. While those commits were not directly rebased into this repository, the same (or similar) changes were made in spirit and since we already got the signoffs from those individuals, I think it's worth leaving in some mention of that, just to cover our bases. 

I'll update this sentence to focus more on those contributions than the LLVM backports.